ts/components/CiePinpad/index.tsx

```diff
@@ -0,0 +1,191 @@
+  codeSplit: ReadonlyArray<string>;
+};
+
+const styles = StyleSheet.create({
+  placeholderContainer: {
+    flexDirection: "row",
+    justifyContent: "center",
+    backgroundColor: variables.colorWhite
+  },
+  text: {
+    alignSelf: "center",
+    justifyContent: "center",
+    color: variables.colorWhite
+  },
+  placeHolderStyle: {
+    height: 4,
+    marginLeft: 2,
+    marginRight: 2,
+    marginTop: 2
+  }
+});
+
+const CIE_PIN_LENGHT = 8;
+/**
+ * A customized CodeInput component.
+ */
+class CiePinpad extends React.PureComponent<Props, State> {
+  private inputs: ReadonlyArray<TextInput>;
+
+  constructor(props: Props) {
+    super(props);
+    this.inputBoxGenerator = this.inputBoxGenerator.bind(this);
+    this.inputs = [];
+    this.state = {
+      pin: new Array(this.props.pinLength).fill(""),
+      pinSelected: 0,
+      codeSplit: []
+    };
+  }
+
+  private handleOnChangeText = (text: string, index: number) => {
+    const pin = this.updatePin(text, index);
+    this.props.onPinChanged(pin.join(""));
+    // if it is not the last element, change focus on next element
+    // handleOnKeyPress is used to handle the Backspace press
+    if (index + 1 < this.inputs.length) {
+      this.inputs[index + 1].focus();
+      return;
+    }
+  };
+
+  // tslint:disable-next-line: readonly-array
+  private updatePin = (text: string, index: number): string[] => {
+    // tslint:disable-next-line: readonly-array
+    const tempPin = [...this.state.pin];
+    tempPin.splice(index, 1, text);
+    const pin: ReadonlyArray<string> = tempPin;
+    this.setState({ pin });
+    return tempPin;
+  };
+
+  private handleOnKeyPress = (
+    nativeEvent: TextInputKeyPressEventData,
+    index: number
+  ) => {
+    if (nativeEvent.key === "Backspace") {
+      // if it is the first element, do nothing
+      if (index === 0) {
+        return;
+      }
+      // if (index !== 0) {
+      //   this.inputs[index - 1].focus(); // it change the focus on the previous input
+      //   // if (this.props.onDeleteLastDigit) {
+      //   //   this.props.onDeleteLastDigit();
+      //   // }
+      // }
+      // check if a deletion is going.
+
+      if (!this.state.pin[index] || this.state.pin[index].length === 0) {
+        this.updatePin("", index - 1);
+        this.inputs[index - 1].focus();
+      }
+      return;
+    }
+  };
+
+  private inputBoxGenerator = (i: number) => {
+    const width = 36;
+    const margin = 2;
+    const totalMargins = margin * 2 * (this.props.pinLength - 1);
+    const screenWidth = Dimensions.get("window").width;
+    const widthNeeded = width * this.props.pinLength + totalMargins;
+
+    // if we have not enough space to place inputs
+    // compute a new width to fit it
+    // consider margin from both sides too
+
+    const targetDimension =
+      widthNeeded > screenWidth
+        ? (screenWidth - totalMargins) / this.props.pinLength
+        : width;
+
+    // tslint:disable-next-line:no-commented-code
```

Comment:
?

Reply:
I put it because in a previous version I added a comment. Now I remove it